src/tests/test_database.py

```diff
@@ -0,0 +1,294 @@
+            db_key = name if name else src.name
+            self.dbs[db_key] = UnitTestDB(server_url, src_path, name)
+            # Check that the database has been created correctly
+            assert self.dbs[db_key], "UnitTestDB should not be empty"
+            assert self.dbs[db_key].dbc, "UnitTestDB's database connection should not be empty"
+            # Check that the database has been loaded correctly from the dump files
+            result = self.dbs[db_key].dbc.execute("SELECT * FROM gibberish")
+            assert len(result.fetchall()) == 6, "Unexpected number of rows found in 'gibberish' table"
+
+    @pytest.mark.parametrize(
+        "db_key",
+        [
+            param('mock_db', marks=pytest.mark.dependency(depends=['init'], scope='class')),
+            param('renamed_db', marks=pytest.mark.dependency(depends=['init_renamed'], scope='class')),
+        ],
+    )
+    def test_drop(self, db_key: str) -> None:
+        """Tests that the previously created object :class:`UnitTestDB` is dropped correctly.
+
+        Args:
+            db_key: Key assigned to the UnitTestDB created in :meth:`TestUnitTestDB.test_init()`.
+
+        """
+        self.dbs[db_key].drop()
+        if self.dbs[db_key].dbc.dialect == 'sqlite':
+            # For SQLite databases, just check if the database file still exists
+            assert not Path(self.dbs[db_key].dbc.db_name).exists(), "The database file has not been deleted"
+        else:
+            with raises(OperationalError, match=r'Unknown database'):
+                self.dbs[db_key].dbc.execute("SELECT * FROM gibberish")
+
+
+@pytest.mark.parametrize("database", [{'src': 'mock_db'}], indirect=True)
+class TestDBConnection:
+    """Tests :class:`DBConnection` class.
+
+    Attributes:
+        dbc (DBConnection): Database connection to the unit test database.
+        server (str): Server url where the unit test database is hosted.
+
+    """
+
+    dbc = None  # type: DBConnection
+    server = None  # type: str
+
+    # autouse=True makes this fixture be executed before any test_* method of this class, and scope='class' to
+    # execute it only once per class parametrization
+    @pytest.fixture(scope='class', autouse=True)
+    def setup(self, request: FixtureRequest, database: UnitTestDB) -> None:
+        """Loads the required fixtures and values as class attributes.
+
+        Args:
+            request: Access to the requesting test context.
+            database: Unit test database (fixture).
+
+        """
+        # Use type(self) instead of self as a workaround to @classmethod decorator (unsupported by pytest and
+        # required when scope is set to "class" <https://github.com/pytest-dev/pytest/issues/3778>)
+        type(self).dbc = database.dbc
+        type(self).server = request.config.getoption('server')
+
+    @pytest.mark.dependency(name='test_init', scope='class')
+    def test_init(self) -> None:
```

Comment:
Quick question, why make a test on which depends every single other subsequent tests, instead of moving this statement in the setup?

Reply:
`setup` is a fixture, so it shouldn't contain any tests. I have tried to find a different alternative that would avoid that but haven't been able to find it. Maybe @luca-drf has more information about this?

Reply:
I was not talking about setup.py file. Sorry for the misunderstanding, but I mean the setupClass method, or is it not existing in pytest?

Reply:
I re-read your message and now understand what you mean :-). Sorry for the confusion, still I find it very weird to have dozens of tests run at class scope level every time we run a test. to check if attributes are set as expected. If any of these tested vars is not set, the tests would fail anyway.